src/Loader/Http/Cookies/CookieJar.php

```diff
@@ -123,37 +123,42 @@ protected function getForDomainFromUrl(string|UriInterface|Url $url): ?string
 
     protected function buildSetCookieHeaderFromBrowserCookie(BrowserCookie $cookie): string
     {
-        $header = $cookie->getName() . '=' . $cookie->getValue();
-
-        if ($cookie->getDomain() !== null) {
-            $header .= '; Domain=' . $cookie->getDomain();
-        }
-
-        if ($cookie->offsetExists('expires') && $cookie->offsetGet('expires') !== -1) {
-            $header .= '; Expires=' . $this->formatExpiresValue($cookie->offsetGet('expires'));
-        }
-
-        if ($cookie->offsetExists('max-age') && !empty($cookie->offsetGet('max-age'))) {
-            $header .= '; Max-Age=' . $cookie->offsetGet('max-age');
-        }
-
-        if ($cookie->offsetExists('path') && !empty($cookie->offsetGet('path'))) {
-            $header .= '; Path=' . $cookie->offsetGet('path');
-        }
+        $attributes = [
+            'domain' => 'Domain',
+            'expires' => 'Expires',
+            'max-age' => 'Max-Age',
+            'path' => 'Path',
+            'secure' => 'Secure',
+            'httpOnly' => 'HttpOnly',
+            'sameSite' => 'SameSite',
+        ];
+
+        $parts = [sprintf('%s=%s', $cookie->getName(), $cookie->getValue())];
+
+        foreach ($attributes as $name => $setCookieName) {
+            $setCookieValue = $cookie->offsetGet($name);
+            if (empty($setCookieValue)) {
```

Comment:
`empty` is one of the bad parts of PHP, it is too complex.
It handles: null, empty string, false.